### PR TITLE
[IMT-372]: Initial changes to support the SONiC on the IMT BCM.

### PR DESCRIPTION
**Pull request by YuriiHordynskyi**
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- Why I did it**
To add the SONiC support on the IMT Broadcom line of devices.

**- How I did it**
Extended the SONiC platform/device-specific modules with the IMT Broadcom line of devices.

**- How to verify it**
Install and run the SONiC on the IMT Broadcom line of devices.

**- Description for the changelog**
* Add the "im-2948-6xl" device configurations.
* Modified the ONIE installer script to support both the IMT Broadcom and Barefoot lines of devices.
* https://niagara-networks.atlassian.net/browse/IMT-372
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

![20221022_142626](https://user-images.githubusercontent.com/48444208/197801784-ca0fdc08-0b7f-4a58-807b-c919939e9bf2.jpg)


```
admin@sonic:~$ show version                                                                       
SONiC Software Version: SONiC.202106-sonic-on-bcm.0-dirty-20221025.113913
Distribution: Debian 10.13
Kernel: 4.19.0-12-2-amd64
Build commit: 2da85f34     
Build date: Tue Oct 25 12:16:26 UTC 2022                                                                                                        
Built by: yuriih@IMT-UA-MONSTER-Server                                                                                                        
                                                                                                                                              
Platform: x86_64-im_n29xx_t40n-r0                                                                                                             
HwSKU: IM2948P                                                                                                                                 
ASIC: broadcom                                                                                                                                 
ASIC Count: 1                                                                                                                                 
Serial Number: None                                                                                                                          
Model Number: None                                                                                                                            
Hardware Revision: None                                                                                                                           
Uptime: 19:33:53 up 40 min,  1 user,  load average: 2.19, 2.27, 2.22
                                                                         
Docker images:            
REPOSITORY                 TAG                                           IMAGE ID            SIZE
docker-dhcp-relay          latest                                        6f1da581efdb        432MB
docker-syncd-brcm          202106-sonic-on-bcm.0-dirty-20221025.113913   5f834fa270f5        721MB
docker-syncd-brcm          latest                                        5f834fa270f5        721MB
docker-snmp                202106-sonic-on-bcm.0-dirty-20221025.113913   ff176692b671        468MB
docker-snmp                latest                                        ff176692b671        468MB
docker-teamd               202106-sonic-on-bcm.0-dirty-20221025.113913   d1f18f44f370        437MB
docker-teamd               latest                                        d1f18f44f370        437MB
docker-platform-monitor    202106-sonic-on-bcm.0-dirty-20221025.113913   d41ffef186d7        641MB
docker-platform-monitor    latest                                        d41ffef186d7        641MB
docker-lldp                202106-sonic-on-bcm.0-dirty-20221025.113913   196529f5af05        465MB
docker-lldp                latest                                        196529f5af05        465MB
docker-orchagent           202106-sonic-on-bcm.0-dirty-20221025.113913   f637333b1426        455MB
docker-orchagent           latest                                        f637333b1426        455MB
docker-database            202106-sonic-on-bcm.0-dirty-20221025.113913   6c8e0453d2f4        425MB
docker-database            latest                                        6c8e0453d2f4        425MB
docker-router-advertiser   202106-sonic-on-bcm.0-dirty-20221025.113913   9d9cae1c8baa        425MB
docker-router-advertiser   latest                                        9d9cae1c8baa        425MB
docker-fpm-frr             202106-sonic-on-bcm.0-dirty-20221025.113913   5968c2af0ce8        455MB
docker-fpm-frr             latest                                        5968c2af0ce8        455MB
                                                                                                  
admin@sonic:~$    
```

### CODE REVIEW

File: platform/broadcom/platform-modules-imt.mk [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# BCM IMT platform modules
+
+BCM_IMT_BK_VERSION = 3.01
+
+export BCM_IMT_BK_VERSION
+
+BCM_IMT_BK_MODULES = boardkeeper-modules_$(BCM_IMT_BK_VERSION)_amd64.deb
+#$(BCM_IMT_BK_MODULES)_URL = "https://github.com/InterfaceMasters/sonic-release-pkgs/raw/imt-bf-sde-9.6.0/sonic-imt-boardkeeper-pkgs/$(BCM_IMT_BK_MODULES)"
+$(BCM_IMT_BK_MODULES)_URL = "http://192.168.157.83:8080/yuriih/im-broadcom-sonic/$(BCM_IMT_BK_MODULES)"
```
Comment (YuriiHordynskyi) [inline platform/broadcom/platform-modules-imt.mk]: Replace by https://github.com/InterfaceMasters/sonic-release-pkgs!!!

File: build_image.sh [hunk 2]
```diff
@@ -35,7 +35,11 @@ generate_onie_installer_image()
             if [ -f ./device/$VENDOR/$PLATFORM/installer.conf ]; then
                 cp ./device/$VENDOR/$PLATFORM/installer.conf ./installer/x86_64/platforms/$PLATFORM
             fi
-
+            if [ $VENDOR == "imt" ]; then
+                if [ -f ./device/$VENDOR/$PLATFORM/${TARGET_MACHINE}_installer_conf/installer.conf ]; then
+                    cp ./device/$VENDOR/$PLATFORM/${TARGET_MACHINE}_installer_conf/installer.conf ./installer/x86_64/platforms/$PLATFORM
+                fi
+            fi
```
Comment (YuriiHordynskyi) [inline build_image.sh:42]: Copying is working during the build procedure:
```
+ '[' -f ./device/imt/x86_64-im_n29xx_t40n-r0/installer.conf ']'
+ '[' imt == imt ']'
+ '[' -f ./device/imt/x86_64-im_n29xx_t40n-r0/broadcom_installer_conf/installer.conf ']'
+ cp ./device/imt/x86_64-im_n29xx_t40n-r0/broadcom_installer_conf/installer.conf ./installer/x86_64/platforms/x86_64-im_n29xx_t40n-r0

```

**Review by oleksandrn-imt — APPROVED** [review]

**Review by AndriiKit — APPROVED** [review]
Looks good, especially a cat :) Yours ?

**Comment by YuriiHordynskyi** [comment]
> Looks good, especially a cat :) Yours ?

yes. It is Alisa. also, I have Sonya and Nick (total - 3 cats).
you will see them in my future PRs=)

File: device/imt/x86_64-im_n29xx_t40n-r0/IM2948P/td2-im2948p-48x10G-6x40G.config.bcm [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+portmap_67.0=83:10:i
+portmap_68.0=84:10:i
+
+#QSFP+ 54
+portmap_69.0=85:40
+portmap_70.0=86:10:i
+portmap_71.0=87:10:i
+portmap_72.0=88:10:i
+#
+# End of IM2948P board
```
Comment (vitalivanov) [inline device/imt/x86_64-im_n29xx_t40n-r0/IM2948P/td2-im2948p-48x10G-6x40G.config.bcm:145]: so, you are adding one platform support for now... but our target at the end would be to have approach with boardutil that is generating a boards.txt->config.bcm and we can load same image to different hw?

Comment (YuriiHordynskyi) [reply device/imt/x86_64-im_n29xx_t40n-r0/IM2948P/td2-im2948p-48x10G-6x40G.config.bcm:145]: It is device related configuration. I will modify it and left only sai related config. Port configuration I am planning to take from the boards.txt. meanwhile, I followed the SONIC approach and added the device-specific config.bcm file.
During the build, there is a check for that file's presence and sai-related configuration options. For modular and hybrid devices (4248-6c, 4540) I have a plan for how to handle that based on the boards.txt file.

**Review by vitalivanov — APPROVED** [review]

**Pull request merged**